modules/common/tls/tls.go

```diff
@@ -37,3 +50,81 @@ type TLS struct {
 	Service *Service `json:"service"`
 	Ca      *Ca      `json:"ca"`
 }
+
+// NewTLS - initialize and return a TLS struct
+func NewTLS(ctx context.Context, cr client.Object, clientset *kubernetes.Clientset, scheme *runtime.Scheme, namespace string, helperInstance *helper.Helper, service *Service, ca *Ca) (*TLS, error) {
```

Comment:
we don't need to pass in the cr, clientset and scheme its not used, right?

```
func NewTLS(ctx context.Context, namespace string, helperInstance *helper.Helper, service *Service, ca *Ca) (*TLS, error) {
```

Reply:
You're right, they're remnants from the previous definition. Pushing correction in a moment.